ivy_tests/test_ivy/test_frontends/test_numpy/test_mathematical_functions/test_miscellaneous.py

```diff
@@ -370,6 +370,49 @@ def test_numpy_gcd(
     )
 
 
+# gradient
+@handle_frontend_test(
+    fn_tree="numpy.gradient",
+    dtype_input_axis=helpers.dtype_values_axis(
+        available_dtypes=("float32", "float16", "float64"),
```

Comment:
the available dtypes should be a strategy that gets all valid dtypes, you should then set the supported dtypes for the function using one of the decorators specified here https://unify.ai/docs/ivy/overview/deep_dive/data_types.html#supported-and-unsupported-data-types